cachelib/allocator/tests/MMTypeTest.h

```diff
@@ -231,26 +234,31 @@ void MMTypeTest<MMType>::testRemoveBasic(Config config) {
   }
 
   std::set<int> foundNodes;
-  for (auto itr = c.getEvictionIterator(); itr; ++itr) {
-    foundNodes.insert(itr->getId());
-  }
+  c.withEvictionIterator([&foundNodes](auto&& itr) {
+    while (itr) {
+      foundNodes.insert(itr->getId());
+      ++itr;
+    }
+  });
 
   for (const auto& node : removedNodes) {
     ASSERT_EQ(foundNodes.find(node->getId()), foundNodes.end());
   }
   // trying to remove through iterator should work as expected as well.
   // no need of iter++ since remove will do that.
-  for (auto iter = c.getEvictionIterator(); iter;) {
-    auto& node = *iter;
-    ASSERT_TRUE(node.isInMMContainer());
-
-    // this will move the iter.
-    c.remove(iter);
-    ASSERT_FALSE(node.isInMMContainer());
-    if (iter) {
-      ASSERT_NE((*iter).getId(), node.getId());
+  c.withEvictionIterator([&foundNodes, &c](auto&& iter) {
```

Comment:
I think this is based on old version of CS-patch PR. In the newest one I have rolled-back those changes to tests to limit number of lines changed.  You can see here (https://github.com/facebook/CacheLib/pull/182/files#diff-5298053236abd39332070ce33eda4db9571ef064728bfe48bbf1747306c32997) that I just extended the existing tests with verifyIterationVariants